### PR TITLE
chore: remove relay server from tensorboard tests

### PR DESCRIPTION
**Pull request by jacobromero**
Description
-----------

What does the PR do? Include a concise description of the PR contents.

This PR migrates tensorboard tests from using the relay_server to use the new wandb_backend_spy fixture.

- [x] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?
- nox -s "system_tests" --python "3.12" -- tests/system_tests/test_functional/test_tensorboard/
  - `= 38 passed, 2 skipped, 26 warnings in 90.64s (0:01:30) =`

### CODE REVIEW

File: tests/system_tests/test_functional/test_tensorboard/test_tf_summary.py [hunk 1]
```diff
@@ -239,50 +231,54 @@ def test_compat_tensorboard(relay_server, wandb_init):
...
+        assert mock_wandb_log.warned(
+            "Step cannot be set when using tensorboard syncing"
+        )
+        history = snapshot.history(run_id=run.id)
+        assert len(history) == 11
+
+        history_df = pd.DataFrame.from_dict(history, orient="index")
+
+        assert history_df["x_scalar"].dropna().tolist() == [i**2 for i in range(10)]
+        assert history_df["y_scalar"].dropna().tolist() == [1337]
```
Comment (timoffex) [inline tests/system_tests/test_functional/test_tensorboard/test_tf_summary.py:274]: Here, ideally we would do something like:

```python
assert history == {
    0: {"x_scalar": ..., "global_step": ...},
    1: ...
}
```

but unfortunately we can't predict the order in which `run.log()` happens relative to the TB event processing logic. Still, if you only call `tf.summary.scalar()` once, you could check that `history` is one of two possibilities:

```python
x_scalar_row = {"x_scalar": ..., "global_step": ...}
y_scalar_row = {"y_scalar": 1337}
assert history in (
    {0: x_scalar_row, 1: y_scalar_row},
    {0: y_scalar_row, 1: x_scalar_row},
)
```

(I think that syntax should work, but I haven't checked)

And I think you can drop the `summary` check; we don't emit summary records from the TB integration, and the codepath that computes summary values is thoroughly tested elsewhere.

File: tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py [hunk 2]
```diff
@@ -13,109 +13,104 @@
...
         assert summary["global_step"] == 99
         assert summary["y=2x"] == 99 * 2
 
-        history = relay.context.get_run_history(run_id)
+        history = snapshot.history(run_id=run.id)
         assert len(history) == 100
 
-        telemetry = relay.context.get_run_telemetry(run_id)
+        config = snapshot.config(run_id=run.id)
+        telemetry = config["_wandb"]["value"]["t"]
```
Comment (timoffex) [inline tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py:34]: `snapshot.telemetry()`

File: tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py [hunk 3]
```diff
@@ -13,109 +13,104 @@
...
-        summary = relay.context.get_run_summary(run_id)
+        summary = snapshot.summary(run_id=run.id)
         assert summary["value_one/value"] == pytest.approx(1.1)
         assert summary["value_one/global_step"] == 9
         assert summary["value_two/value"] == pytest.approx(2.2)
         assert summary["value_two/global_step"] == 9
 
-        telemetry = relay.context.get_run_telemetry(run_id)
+        config = snapshot.config(run_id=run.id)
+        telemetry = config["_wandb"]["value"]["t"]
```
Comment (timoffex) [inline tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py:63]: `snapshot.telemetry()` (just marking all these so none are missed)

File: tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py [hunk 4]
```diff
@@ -13,109 +13,104 @@
...
         assert summary["global_step"] == 10
 
         assert summary["example"]["_type"] == "images/separated"
         assert summary["example"]["width"] == 28
         assert summary["example"]["height"] == 28
         assert summary["example"]["format"] == "png"
 
-        telemetry = relay.context.get_run_telemetry(run_id)
+        config = snapshot.config(run_id=run.id)
+        telemetry = config["_wandb"]["value"]["t"]
```
Comment (timoffex) [inline tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py:91]: `snapshot.telemetry()`

File: tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py [hunk 5]
```diff
@@ -125,56 +120,55 @@ def test_add_gif(wandb_init, relay_server):
...
         assert summary["global_step"] == 0
         assert summary["my_image_batch"]["_type"] == "images/separated"
         assert summary["my_image_batch"]["width"] == 800
         assert summary["my_image_batch"]["height"] == 200
         assert summary["my_image_batch"]["count"] == 1
         assert summary["my_image_batch"]["format"] == "png"
 
-        telemetry = relay.context.get_run_telemetry(run_id)
+        config = snapshot.config(run_id=run.id)
+        telemetry = config["_wandb"]["value"]["t"]
```
Comment (timoffex) [inline tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py:146]: `snapshot.telemetry()`

File: tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py [hunk 6]
```diff
@@ -125,56 +120,55 @@ def test_add_gif(wandb_init, relay_server):
...
+        assert len(snapshot.run_ids()) == 1
+
+        summary = snapshot.summary(run_id=run.id)
         assert summary["global_step"] == 4
         assert summary["distribution centers"]["_type"] == "histogram"
         assert len(summary["distribution centers"]["values"]) == 500
 
-        telemetry = relay.context.get_run_telemetry(run_id)
+        config = snapshot.config(run_id=run.id)
+        telemetry = config["_wandb"]["value"]["t"]
```
Comment (timoffex) [inline tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py:171]: `snapshot.telemetry()`

File: tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py [hunk 7]
```diff
@@ -183,46 +177,44 @@ def test_add_histogram(wandb_init, relay_server):
...
+        assert len(snapshot.run_ids()) == 1
 
-        summary = relay.context.get_run_summary(run_id)
+        summary = snapshot.summary(run_id=run.id)
         assert summary["pr_curve_table"]["_type"] == "table-file"
         assert summary["pr_curve_table"]["ncols"] == 2
 
-        telemetry = relay.context.get_run_telemetry(run_id)
+        config = snapshot.config(run_id=run.id)
+        telemetry = config["_wandb"]["value"]["t"]
```
Comment (timoffex) [inline tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py:195]: `snapshot.telemetry()`

File: tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py [hunk 8]
```diff
@@ -183,46 +177,44 @@ def test_add_histogram(wandb_init, relay_server):
...
+    with wandb_backend_spy.freeze() as snapshot:
+        assert len(snapshot.run_ids()) == 1
 
-        summary = relay.context.get_run_summary(run_id)
+        summary = snapshot.summary(run_id=run.id)
         assert summary["pr_curve"]["_type"] == "table-file"
         assert summary["pr_curve"]["ncols"] == 2
 
-        telemetry = relay.context.get_run_telemetry(run_id)
+        config = snapshot.config(run_id=run.id)
```
Comment (timoffex) [inline tests/system_tests/test_functional/test_tensorboard/test_torch_tb.py:216]: `snapshot.telemetry()`

**Review by timoffex — APPROVED** [review]
Thank you!

File: tests/system_tests/test_functional/test_tensorboard/test_tf_summary.py [hunk 9]
```diff
@@ -124,97 +119,94 @@ def test_batch_images(wandb_init, relay_server):
...
         assert summary["loss"] == pytest.approx(scalars[-1])
 
-        history = relay.context.get_run_history(run.id)
-        assert len(history) == 3
-        assert history["loss"].tolist() == pytest.approx(scalars)
+        history = snapshot.history(run_id=run.id)
+        history_df = pd.DataFrame.from_dict(history, orient="index")
+
+        assert len(history_df) == 3
+        assert history_df["loss"].tolist() == pytest.approx(scalars)
```
[the remark this answers is not in the document]
Comment (jacobromero) [reply tests/system_tests/test_functional/test_tensorboard/test_tf_summary.py:140]: it looks like there is possibly some floating point error when writing via tensorboard.

running without `pytest.approx` gives the following error
```
assert 0.3449999988079071 == 0.345
```

I will make the other change to git rid of using pandas dataframe though

**Review by timoffex — APPROVED** [review]

**Pull request merged**